packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -170,42 +180,43 @@ library LibProposing {
         // Validate the prover assignment, then charge Ether or ERC20 as the
         // prover fee based on the block's minTier.
         uint256 proverFee =
-            _validateAssignment(blk.minTier, txListHash, assignment);
+            _validateAssignment(blk.minTier, blk.blobHash, assignment);
 
         emit BlockProposed({
             blockId: blk.blockId,
             assignedProver: blk.assignedProver,
             livenessBond: config.livenessBond,
             proverFee: proverFee,
             minTier: blk.minTier,
-            meta: meta
+            meta: meta,
+            usingBlob: blk.usingBlob
         });
     }
 
     /// @dev Hashing the block metadata.
     function hashMetadata(TaikoData.BlockMetadata memory meta)
         internal
         pure
-        returns (bytes32 hash)
+        returns (bytes32)
     {
-        uint256[7] memory inputs;
-        inputs[0] = uint256(meta.l1Hash);
-        inputs[1] = uint256(meta.difficulty);
-        inputs[2] = uint256(meta.txListHash);
-        inputs[3] = uint256(meta.extraData);
-        inputs[4] = (uint256(meta.id)) | (uint256(meta.timestamp) << 64)
-            | (uint256(meta.l1Height) << 128) | (uint256(meta.gasLimit) << 192);
-        inputs[5] = uint256(uint160(meta.coinbase));
-        inputs[6] = uint256(keccak256(abi.encode(meta.depositsProcessed)));
-
-        assembly {
-            hash := keccak256(inputs, 224 /*mul(7, 32)*/ )
-        }
+        return keccak256(
```

Comment:
@smtmfft please review this change. If you still prefer the old approach let me know.